### ISSUE TITLE
CPU hours and Games played on the Users page increase too fast.

### ISSUE THREAD

**Issue by gvreuls** [glinscott/fishtest#402]
Since a few days the CPU hours and Games played numbers on the [Users page](http://tests.stockfishchess.org/users) are growing dramatically. For reference, at September the 21st of 2019 my stats showed 153400 CPU hours and 13000000 games played, while as of writing it's grown to 183400 CPU hours and 15500000 games played. Usually my workers produce about 10000 CPU hours and 850000 games per month, so the users page is indicating about 3 months of gain in 5 days.

This might be related to #397 which fixes similar issues for the games per hour stats.

**Comment by ppigazzini** [comment]
@gvreuls thank you for the alert. https://github.com/glinscott/fishtest/pull/399 should have fixed the problem. Anyway the values are computed from the data stored in mongodb, the test of the new script is not destructive.

**Comment by gvreuls** [comment]
The numbers look more reasonable now, but all users seem to have lost a few thousand CPU hours (and games played). This isn't a big problem for me personally but people with slower workers seem to have lost months of work; horst.prack is a good example, he has less CPU hours now than in the [Top CPU Contributors list](https://github.com/official-stockfish/Stockfish/blob/master/Top%20CPU%20Contributors.txt) of 11 months ago.

**Comment by ppigazzini** [comment]
@gvreuls thank you for the help
cc @tomtor

**Comment by tomtor** [comment]
@gvreuls In the last commit I changed https://github.com/glinscott/fishtest/blob/c09c1a5d765b9ca1c051845e7e0a15898bb80353/fishtest/utils/delta_update_users.py#L39
It used to assign the task size (250 now, 1000 in the past) when no games were recorded. I cannot think of a reason to do that, so I changed it to zero. If anybody has a suggestion why the original code is correct I would like to hear about it.

**Comment by ppigazzini** [comment]
@tomtor wow, so the values were inflated.

**Comment by gvreuls** [comment]
@tomtor Thanks for the info. It may be prudent to announce this change and explain the deflated stats on the forum so users know what's going on?

**Comment by tomtor** [comment]
> am I right in the assumption that before this patch, users could artificially inflate their stats by stopping their worker after it started playing a batch of games but before it finished the first one?

@gvreuls Yes, in theory that would be possible, but this also happens in normal operation. Especially slow workers will after compiling often be signaled that the run has already finished.

I am not sure we need to report this on the forum, because most users will not notice the change or may follow the fishtest github issues. If we *do* report it on the forum I would like to wait a while till we are sure it is 100% correct.

**Comment by Vizvezdenec** [comment]
can this stuff be manually reupdated (at least for tests number)?
I mean just go thru all people with at least 1 test and update number that is shown by fishtest web page
tests.stockfishchess.org/tests/user/Vizvezdenec for me it should be 6078 tests and not 7000+ :)

**Comment by tomtor** [comment]
@Vizvezdenec https://github.com/glinscott/fishtest/pull/416 should prevent this from happening in the future, but the statistics indeed need to be rebuild (`delta_update_users.py full`) by @ppigazzini at a convenient time, when the periodic crontab is disabled.

**Comment by ppigazzini** [comment]
@tomtor merged https://github.com/glinscott/fishtest/pull/416, done full scan:
```
$ time python delta_update_users.py full
scan all
Warning: Found old user to delete: 5c4656b00ebc5902bb5d6595
Warning: Found old user to delete: 5c0b4e600ebc5902bceebeeb
Warning: Found old user to delete: 5d2b3bf50ebc5925cf0d9f46
...
Warning: Found old user to delete: 5bfccec70ebc5902bced861c
Successfully updated 1312 users

real    7m16.709s
user    3m21.294s
sys     0m3.610s
```

**Comment by tomtor** [comment]
@ppigazzini Thanks, stats look good now!
